webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +87,1137 @@ public WebSiteSessionController createComponentSessionController(
-          // Filled at first access then null
-          String nomPage = request.getParameter("nomPage");
-          String tempPopup = request.getParameter("popup");
-          // Filled at first creation then null
-          String listeIcones = request.getParameter("ListeIcones");
-          // = en cas de new Site ou de classifySite
-          String listeTopics = request.getParameter("ListeTopics");
-          // Retrieve positions
-          String positions = request.getParameter("Positions");
-
-          int popup = 0;
-          if ((tempPopup != null) && (tempPopup.length() > 0)) {
-            popup = 1;
-          }
+    Collection<SiteDetail> listeSites = scc.getAllWebSite();
+    request.setAttribute("ListSites", listeSites);
+    request.setAttribute("CurrentFolder", scc.getSessionTopic());
 
-          ArrayList<String> listIcons = new ArrayList<>();
-          int begin = 0;
-          int end = 0;
-          if (listeIcones != null) {
-            end = listeIcones.indexOf(',', begin);
-            while (end != -1) {
-              listIcons.add(listeIcones.substring(begin, end));
-              begin = end + 1;
-              end = listeIcones.indexOf(',', begin);
-            }
-          }
+    return
+        "/webSites/jsp/classifyDeclassify.jsp?Action=" + action + "&TopicId=" + id + "&Path=" +
+            linkedPathString;
+  }
 
-          /* recuperation de l'id */
-          id = scc.getNextId();
+  private String updateTopic(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
 
-          /* Creer le repertoire id */
-          scc.createFolder(scc.getWebSitePathById(id));
+    String id = request.getParameter("ChildId");
+    String path = request.getParameter("Path");
 
-          // Persist siteDetail inside database type 0 = site cree
-          SiteDetail descriptionSite =
-              new SiteDetail(id, scc.getComponentId(), nomSite, description, nomPage, 0, null, null,
-                  0, popup);
-          descriptionSite.setPositions(positions);
+    NodeDetail folderDetail = scc.getFolderDetail(id);
+    request.setAttribute("CurrentFolder", folderDetail);
 
-          String pubId = scc.createWebSite(descriptionSite);
-          descriptionSite = scc.getWebSite(id);
-          scc.setSessionSite(descriptionSite);
+    return "/webSites/jsp/updateTopic.jsp?ChildId=" + id + "&Path=" + path;
+  }
 
-          if (listIcons.size() > 0) {
-            scc.associateIcons(id, listIcons);
-          }
+  private String addTopic(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
 
-          currentPath = scc.getWebSitePathById(id);
+    String action = request.getParameter("Action");// =Add
+    String fatherId = request.getParameter("Id");
+    String newTopicName = request.getParameter("Name");
+    String newTopicDescription = request.getParameter("Description");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Description" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)